src/test/java/net/greghaines/jesque/IntegrationTest.java

```diff
@@ -217,6 +229,24 @@ private static void assertMixed(final WorkerListener listener, final WorkerEvent
         }
     }
 
+    private static void assertBatchMixed(final WorkerListener listener, final WorkerEvent... events) {
+        final Job job1 = new Job("FailAction");
+        final Job job2 = new Job("TestAction", new Object[] { 1, 2.3, true, "test", Arrays.asList("inner", 4.5) });
+        final Job job3 = new Job("FailAction");
+        final Job job4 = new Job("TestAction", new Object[] { 1, 2.3, true, "test", Arrays.asList("inner", 4.5) });
+
+        doBatchWork(Arrays.asList(job1, job2, job3, job4),
+                map(entry("FailAction", FailAction.class), entry("TestAction", TestAction.class)), listener, events);
+
+        final Jedis jedis = createJedis(CONFIG);
+        try {
+            Assert.assertEquals("2", jedis.get(createKey(CONFIG.getNamespace(), STAT, FAILED)));
+            Assert.assertEquals("2", jedis.get(createKey(CONFIG.getNamespace(), STAT, PROCESSED)));
+        } finally {
+            jedis.quit();
+        }
+    }
+
     private static void doWork(final List<Job> jobs, final Map<String, ? extends Class<? extends Runnable>> jobTypes,
```

Comment:
one small refactoring that we could do is changing the signature of this method & TestUtils.enqueueJobs to accept a single job and then use it wherever the list of jobs has 1 element.

worker creation could also be extracted to a method

Reply:
I would not do this right now. Let's keep it as it is.